tensorflow/python/framework/dtypes.py

```diff
@@ -136,27 +138,37 @@ def as_datatype_enum(self):
     return self._type_enum
 
   @property
-  def is_integer(self):
-    """Returns whether this is a (non-quantized) integer type."""
-    return (not self.is_quantized and
-            issubclass(self.as_numpy_dtype, np.integer))
+  def is_bool(self):
+    """Returns whether this is a boolean data type"""
+    return self.base_dtype == bool
+
+  @property
+  def is_complex(self):
+    """Returns whether this is a complex floating point type."""
+    return self.base_dtype in (complex64, complex128)
 
   @property
   def is_floating(self):
     """Returns whether this is a (real) floating point type."""
     return issubclass(self.as_numpy_dtype, np.floating)
 
   @property
-  def is_complex(self):
-    """Returns whether this is a complex floating point type."""
-    return self.base_dtype in (complex64, complex128)
```

Comment:
Fine, I do the adjustment cause there are two assigners told me this is a custom of Google, and let me fix it in the previous PRs. So, this time, I adjusted them in passing, while it looks like it's not fit for everyone. I'll revert that~